docs/examples/Experimental.tsx

```diff
@@ -183,73 +183,47 @@ const Option = (props: OptionProps<DateOption, false>) => {
 
 interface DatePickerProps {
   readonly value: DateOption | null;
-  readonly onChange: (value: DateOption | null) => void;
+  readonly onChange: (newValue: DateOption | null) => void;
 }
 
-interface DatePickerState {
-  readonly options: readonly (DateOption | CalendarGroup)[];
-}
+const DatePicker = (props: DatePickerProps) => {
+  const [options, setOptions] = useState(defaultOptions);
 
-class DatePicker extends Component<DatePickerProps, DatePickerState> {
-  state: DatePickerState = {
-    options: defaultOptions,
-  };
-  handleInputChange = (value: string) => {
+  const handleInputChange = (value: string) => {
     if (!value) {
-      this.setState({ options: defaultOptions });
+      setOptions(defaultOptions);
       return;
     }
     const date = chrono.parseDate(suggest(value.toLowerCase()));
-    if (date) {
-      this.setState({
-        options: [createOptionForDate(date), createCalendarOptions(date)],
-      });
-    } else {
-      this.setState({
-        options: [],
-      });
+    if (!date) {
+      setOptions([]);
+      return;
     }
+    setOptions([createOptionForDate(date), createCalendarOptions(date)]);
   };
-  render() {
-    const { value } = this.props;
-    const { options } = this.state;
-    return (
-      <Select<DateOption, false>
-        {...this.props}
-        components={{ Group, Option }}
-        filterOption={null}
-        isMulti={false}
-        isOptionSelected={(o, v) => v.some((i) => i.date.isSame(o.date, 'day'))}
-        maxMenuHeight={380}
-        onChange={this.props.onChange}
-        onInputChange={this.handleInputChange}
-        options={options}
-        value={value}
-      />
-    );
-  }
-}
 
-interface State {
-  readonly value: DateOption | null;
-}
+  return (
+    <Select<DateOption, false>
+      {...props}
+      components={{ Group, Option }}
+      filterOption={null}
+      isMulti={false}
+      isOptionSelected={(o, v) => v.some((i) => i.date.isSame(o.date, 'day'))}
+      maxMenuHeight={380}
+      onChange={props.onChange}
```

Comment:
This is spread on the props that are spread in above

Reply:
This is how it was in the original example. I'm just gonna leave it they way it was. Personally, i think the props spread should go and the `onChange` should stay (it's more explicit), but im too scared to delete the props spread in case i break something 😰